### ISSUE TITLE
Style issues

### ISSUE THREAD

**Issue by raulfdm** [raulfdm/raulmelo.dev#1022]
## Description

Currently when the site is loaded and we navigate to a page, we have a very weird situation where the styles on the post seems not correct:

<img width="600" src="https://user-images.githubusercontent.com/12464600/93735640-ef56aa80-fbdd-11ea-8e52-ab01fca27bb7.gif"/>

In my PR to upgrade dependencies, the situation get even worst:

<img width="600" src="https://user-images.githubusercontent.com/12464600/93735791-7a37a500-fbde-11ea-8cdb-6feeb0ffa268.gif"/>

## Problem

After few hours of investigation I've realized that probably `styled-components` are poorly handling global styles. 

Actually I have 3 ways of styling pages:
1. Blog styles - has `typography.js` styles injected globally
2. CV styles - has its own ways to style things
3. Regular site styles

Blog and Regular styles injects `GlobalStyles` (which contains theme styles) but Blog creates another global style (with `createGlobalStyle` from styled components), passing all `typography.js` and a few custom styles only related to blog.

When we reach `/` (home), we load `GlobalStyles` from regular site styles. When we navigate (via Gatsby link) to a Blog page, it does not load `BlogStyles`. If I then reload the page, now we have `GlobalStyles` and also `BlogStyles` rendered correctly.

## Solution

What I want: **Reload specifically for each page so the other page styles wont have influence on other's page style**

It was VERY tricky to find out a good explanation of how to keep 3 different layouts in Gatsby with global styled components but I've realized that I could use the `wrapPageElement` API to do that.

Basically this API allow us to surround a PAGE with some code and after unmount (change from one page to another), the `wrapPageElement` will be called again.

Since it's a child of Router, I also have access to the location, which means I could predict where I'm (pathname) and based on that, render a Theme which contains either blog, site or cv styles.

**Comment by raulfdm** [comment]
Closed by #1021
